frontend/src/component/changeRequest/ChangeRequest/ChangeRequest.tsx

```diff
@@ -50,37 +60,45 @@ export const ChangeRequest: VFC<IChangeRequestProps> = ({
-                                        enabled={change?.payload?.enabled}
+                                        enabled={change.payload.enabled}
                                         onDiscard={onDiscard(change.id)}
                                     />
-                                }
-                            />
-                            {/* <ConditionallyRender
-                                condition={change.action === 'addStrategy'}
-                                show={
-                                    <StrategyAddedChange>
+                                )}
+                            {change.action === 'addStrategy' &&
+                                hasNameField(change.payload) && (
+                                    <StrategyAddedChange
+                                        onDiscard={onDiscard(change.id)}
+                                    >
                                         <GetFeatureStrategyIcon
                                             strategyName={change.payload.name}
                                         />
+
                                         {formatStrategyName(
-                                            change.payload.name
+                                            change?.payload?.name
                                         )}
                                     </StrategyAddedChange>
-                                }
-                            />
-                            <ConditionallyRender
-                                condition={change.action === 'deleteStrategy'}
-                                show={<StrategyDeletedChange />}
-                            />
-                            <ConditionallyRender
-                                condition={change.action === 'updateStrategy'}
-                                show={<StrategyEditedChange />}
-                            /> */}
+                                )}
+                            {change.action === 'deleteStrategy' && (
+                                <StrategyDeletedChange
+                                    onDiscard={onDiscard(change.id)}
+                                />
+                            )}
+                            {change.action === 'updateStrategy' &&
+                                hasNameField(change.payload) && (
+                                    <StrategyEditedChange
+                                        onDiscard={onDiscard(change.id)}
+                                    >
+                                        <GetFeatureStrategyIcon
+                                            strategyName={change?.payload?.name}
+                                        />
+                                        {formatStrategyName(
+                                            change?.payload?.name
+                                        )}
+                                    </StrategyEditedChange>
+                                )}
```

Comment:
Could we use ConditionallyRender? I prefer the codebase to use the same patterns to not get in a situation where people look at one part of the code and implement it one way, and later look at another part of the code and implement it another way. I'm to discuss how we can make it better, but for now consistency is the key to speed and it's easier to refactor a pattern later if the pattern is the same.

Reply:
I originally started with ConditionallyRender but TypeScript can't infer types in the show and elseShow paths.
E.g. the type guards hasNameField or hasEnabledField are understood by TS in the && condition but not in custom ConditionallyRender component. I'm in two minds here because on the one hand I'd like to stick to the convention on the other hand I'd like to maintain type safety.

Reply:
<img width="575" alt="Screenshot 2022-11-03 at 15 49 17" src="https://user-images.githubusercontent.com/1394682/199753730-55af2a23-715d-42af-a633-7f1aa55727a0.png">
<img width="468" alt="Screenshot 2022-11-03 at 15 49 24" src="https://user-images.githubusercontent.com/1394682/199753753-b7e1e932-150d-4d14-ba8b-43126b98ea21.png">